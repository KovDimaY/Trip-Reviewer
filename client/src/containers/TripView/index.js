import React, { Component } from 'react';
import { getTripWithReviewer, clearTripWithReviewer } from '../../actions';
import { connect } from 'react-redux';

class TripView extends Component {
    componentWillMount(){
        this.props.dispatch(getTripWithReviewer(this.props.match.params.id))
    }

    componentWillUnmount(){
        this.props.dispatch(clearTripWithReviewer())
    }

    renderTrip = (trips) => (
        trips.current 
            ? <div className="br_container">
                    <div className="br_header">
                        <h2>{trips.current.title}</h2>
                        <h5>{trips.current.author}</h5>
                        <div className="br_reviewer">
                            <span>Review by:</span> {trips.reviewer.name} {trips.reviewer.lastname}
                        </div>
                    </div>
                    <div className="br_review">
                        {trips.current.review}
                    </div>
                    <div className="br_box">
                        <div className="left">
                            <div>
                                <span>Duration:</span> {trips.current.duration} days
                            </div>
                            <div>
                                <span>Price:</span> {trips.current.price}
                            </div>
                        </div>
                        <div className="right">
                            <span>Rating</span>
                            <div>{trips.current.rating}/5</div>
                        </div>
                    </div>
                </div>
            :   null
    )

    render() {
        const { trips } = this.props;
        return (
            <div>
                {this.renderTrip(trips)}
            </div>
        );
    }
}

function mapStateToProps(state){
    return {
        trips: state.trips
    };
};

export default connect(mapStateToProps)(TripView);