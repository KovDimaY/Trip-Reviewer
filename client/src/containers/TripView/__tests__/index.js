import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
 
import TripView from './../../TripView';
import Root from './../../../hoc/root';

const mockComponent = props => {
    return (
        <Root>
            <TripView {...props} />
        </Root>
    );
};
 
describe('<TripView />', () => {
    it('should render component without trips', () => {
        const props = {
            trips: {},
            match: {
                params: {
                    id: 'id'
                }
            }
        };
        const tree = create(mockComponent(props)).toJSON();
        
        expect(tree).toMatchSnapshot();
    });
});