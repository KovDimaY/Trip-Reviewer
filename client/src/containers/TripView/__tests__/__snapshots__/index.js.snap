// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripView /> should render component without trips 1`] = `<div />`;
